### ISSUE TITLE
Annotate AnalysisLevel property

### ISSUE THREAD

**Issue by gewarren** [dotnet/docs#29521]
The AnalysisLevel property seems to be shared between C# compiler messages and Roslyn code analysis. I'll log an issue to make note of this in the code analysis and C# compiler options docs.

_Originally posted by @gewarren in https://github.com/dotnet/docs/issues/29498#issuecomment-1130331975_

**Comment by Youssef1313** [comment]
There was a recent change in this area in https://github.com/dotnet/sdk/pull/24826.

Old behavior was that `AnalysisLevel` overwrites `WarningLevel`. So `WarningLevel` wasn't taken into account if `AnalysisLevel` is set.

New behavior is `AnalysisLevel` sets `WarningLevel` only if the user didn't set `WarningLevel` himself.

cc @baronfel @jmarolf @mavasani

**Comment by baronfel** [comment]
Yep, we generally want defaults to only be set if the user hasn't set something, and thanks to @Youssef1313 we found a bug where we were unconditionally setting the WarningLevel. It's definitely been one of those issues' that's spiralled a bit past my initial expected size, though!

**Comment by BillWagner** [comment]
The docs currently describe the old behavior. See [here](https://docs.microsoft.com/dotnet/csharp/language-reference/compiler-options/errors-warnings#warninglevel).

When does the new behavior ship?
